python/adelphi/bin/run-adelphi-tests.py

```diff
@@ -0,0 +1,77 @@
+# Run test suite for some set of Cassandra versions.
+#
+# We implement this as a front-end script because the tox/pytest/unittest chain
+# isn't great about iterating over a suite of test fixtures and re-running the
+# collected tests for each of them.  Rather than fight with the frameworks we
+# manage the fixture iteration manually in this script.  This also has the
+# nice side effect of moving a lot of C* checking/session code out of the test
+# suite, which in turn should allow us to write simpler tests.
```

Comment:
I spent a considerable amount of time trying to find a clear way to manage Cassandra-versions-as-test-fixtures within pytest directly.  There wasn't any obvious way to iterate over a set of fixtures (executing any discovered tests for each) so rather than continue to work against the tooling I moved up a level and implemented the fixture iteration in a front-end script.